Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs

```diff
@@ -292,11 +297,30 @@ class ButtonClickListener : Object, IOnClickListener
 
 			#endregion
 
-			public void OnClick(global::Android.Views.View v)
+			public void OnClick(AView v)
 			{
 				var renderer = v.Tag as ButtonRenderer;
 				((IButtonController)renderer?.Element)?.SendClicked();
 			}
 		}
+
+		class ButtonTouchListener : Object, IOnTouchListener
+		{
+			public static readonly Lazy<ButtonTouchListener> Instance = new Lazy<ButtonTouchListener>(() => new ButtonTouchListener());
+
+			public bool OnTouch(AView v, AMotionEvent e)
+			{
+				var renderer = v.Tag as ButtonRenderer;
+				if (e.Action == AMotionEventActions.Down)
+				{
+					((IButtonController)renderer?.Element)?.SendPressed();
+				}
+				else if (e.Action == AMotionEventActions.Up)
+				{
+					((IButtonController)renderer?.Element)?.SendReleased();
```

Comment:
This is kinda weird casting to IButtonController if it's null it will crash so the '?' isn't really saving you here.